drivers/pds/lib/utils.go

```diff
@@ -549,17 +866,57 @@ func DeployDataServices(supportedDataServicesMap map[string]string, projectID st
 
 				if err != nil {
 					logrus.Warnf("An Error Occured while creating deployment %v", err)
-					return nil, nil, err
+					return nil, nil, nil, err
 				}
 				err = ValidateDataServiceDeployment(deployment)
 				if err != nil {
-					return nil, nil, err
+					return nil, nil, nil, err
 				}
 				deploymentsMap[ds] = append(deploymentsMap[ds], deployment)
 			}
 		}
 	}
-	return deploymentsMap, dataServiceImageMap, nil
+	return deploymentsMap, dataServiceImageMap, dataServiceVersionBuildMap, nil
+}
+
+//UpdateDataServiceVerison modifies the existing deployment version/image
+func UpdateDataServiceVerison(dataServiceID, deploymentID string, appConfigID string, nodeCount int32, resourceTemplateID, dsImage string, dsVersion string) (*pds.ModelsDeployment, error) {
+
+	//Validate if the passed dsImage is available in the list of images
+	var versions []pds.ModelsVersion
+	var images []pds.ModelsImage
+	var dsImageID string
+	versions, err = components.Version.ListDataServiceVersions(dataServiceID)
+	if err != nil {
+		return nil, err
+	}
+	isBuildAvailable = false
+	for i := 0; i < len(versions); i++ {
+		if versions[i].GetName() == dsVersion {
+			images, _ = components.Image.ListImages(versions[i].GetId())
+			for j := 0; j < len(images); j++ {
+				if images[j].GetBuild() == dsImage {
+					dsImageID = images[j].GetId()
+					isBuildAvailable = true
+					break
+				}
+			}
+		}
+	}
+
+	if !(isBuildAvailable) {
+		logrus.Errorf("Version/Build passed is not available")
+	}
+
+	deployment, err = components.DataServiceDeployment.UpdateDeployment(deploymentID, appConfigID, dsImageID, nodeCount, resourceTemplateID, nil)
```

Comment:
do u still want to continue even if build is not available?

Reply:
since i want to fail it in test, I am logging the error in utils and failing it in tests.
is it ok if I fail the test here with logrus.Fatal ?

Reply:
you can either use fatal or return error and fail it in the calling test

Reply:
done